packages/amplify-cli/src/execution-manager.ts

```diff
@@ -217,19 +219,26 @@ const legacyCommandExecutor = async (context: Context, plugin: PluginInfo) => {
   }
 };
 
+const PLUGINS_WITH_HANDLERS = new Set([constants.CORE, constants.CODEGEN]);
```

Comment:
I would name this `EVENT_EMITTING_PLUGINS`. Also, I would just make it an array.